src/miner.h

```diff
@@ -201,6 +202,7 @@ class BlockAssembler
 void IncrementExtraNonce(CBlock* pblock, const CBlockIndex* pindexPrev, unsigned int& nExtraNonce);
 int64_t UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParams, const CBlockIndex* pindexPrev);
 
+// TODO just accept a CBlockIndex*
```

Comment:
I don't understand this TODO. CBlockIndex requires the block to be in the blockchain (block tree). However the coinbase commitment is regenerated *before* the block is submitted to the tree.

Reply:
(Reply https://github.com/bitcoin/bitcoin/pull/21391#issuecomment-812716831)